procsim/biomass/level1_product_generator.py

```diff
@@ -40,47 +55,360 @@
+    PRODUCTS = ['CPF_L1VFRA']
 
     _GENERATOR_PARAMS: List[tuple] = [
+        ('file_class', '_file_class', 'str'),
         ('enable_framing', '_enable_framing', 'bool'),
         ('frame_grid_spacing', '_frame_grid_spacing', 'float'),
         ('frame_overlap', '_frame_overlap', 'float'),
         ('frame_lower_bound', '_frame_lower_bound', 'float'),
         ('slice_overlap_start', '_slice_overlap_start', 'float'),
-        ('slice_overlap_end', '_slice_overlap_end', 'float')
+        ('slice_overlap_end', '_slice_overlap_end', 'float'),
     ]
 
-    def __init__(self, logger, job_config, scenario_config: dict, output_config: dict):
+    def __init__(self, logger, job_config, scenario_config: dict, output_config: dict) -> None:
         super().__init__(logger, job_config, scenario_config, output_config)
         self._enable_framing = True
         self._frame_grid_spacing = constants.FRAME_GRID_SPACING
         self._frame_overlap = constants.FRAME_OVERLAP
-        self._frame_lower_bound = constants.FRAME_LOWER_BOUND
+        self._frame_lower_bound = constants.FRAME_MINIMUM_DURATION
         self._slice_overlap_start = constants.SLICE_OVERLAP_START
         self._slice_overlap_end = constants.SLICE_OVERLAP_END
 
+        self._frame_status = None
+        self._file_class = 'TEST'
+        self._version_nr = 1
+        self._hdr.product_baseline = 0  # This is the default for VFRA
+        self._zip_output = False
+        self._hdr.product_type = 'CPF_L1VFRA'
+
+        self._source_L0S = None
+        self._source_L0M = None
+        self._source_AUX_ORB = None
+
     def get_params(self) -> Tuple[List[tuple], List[tuple], List[tuple]]:
         gen, hdr, acq = super().get_params()
         return gen + self._GENERATOR_PARAMS, hdr + _HDR_PARAMS, acq + _ACQ_PARAMS
 
+    def parse_inputs(self, input_products: Iterable[JobOrderInput]) -> bool:
+        if not super().parse_inputs(input_products):
+            return False
+
+        # Find the L0S, L0M and AUX_ORB products in the input list.
+        source_L0S_files = []
+        source_L0M_files = []
+        source_AUX_ORB_files = []
+        for input in input_products:
+            for file in input.file_names:
+                gen = product_name.ProductName(self._compact_creation_date_epoch)
+                gen.parse_path(file)
+
+                if gen.file_type and L0S_PATTERN.match(gen.file_type):
+                    source_L0S_files.append(file)
+                elif gen.file_type and L0M_PATTERN.match(gen.file_type):
+                    source_L0M_files.append(file)
+                elif gen.file_type == AUX_ORB_TYPE:
+                    source_AUX_ORB_files.append(file)
+                else:
+                    raise ScenarioError(f'Unknown input file type: {gen.file_type} of file {file}')
+
+        # Expect exactly one input file for each product type.
+        if len(source_L0S_files) != 1 or len(source_L0M_files) != 1 or len(source_AUX_ORB_files) != 1:
```

Comment:
By doing this, it is no longer possible to generate such a file without inputs, which is required.
You should add this new product type to test/biomass/procsim_config_generate_all.json, then add scenario parameters to set the required fields, such as the self._source_xxx file names.

Reply:
You're correct. I've removed the check here, since that should not be the responsibility of this method, and added support for partial or absent sets of input files.